source/to_cpp1.h

```diff
@@ -1041,8 +1043,26 @@ class cppfront
     };
     std::vector<arg_info> current_args  = { {} };
 
+    struct active_using_declaration {
+        token const* identifier = {};
+
+        explicit active_using_declaration(using_statement_node const& n) {
+          if (auto id = get_if<id_expression_node::qualified>(&n.id->id)) {
+              identifier = (*id)->ids.back().id->identifier;
+          }
+        }
+    };
+
+    using unqualified_name_lookup_res =
+        std::optional<std::variant<declaration_node const*, active_using_declaration>>;
+
+    //  Stack of the currently active nested declarations we're inside
     std::vector<declaration_node const*> current_declarations = { {} };
 
+    //  Stack of the currently active nested names we're inside:
+    //  Like 'current_declarations' + also parameters and using declarations
+    std::vector<unqualified_name_lookup_res::value_type> current_names = { {} };
```

Comment:
I don't think "active nested names" is enough to describe that `current_names`
keeps the names active while they're in scope.

Everything else looks fine.

Reply:
This is what I think you need to change:
```diff
-Stack of the currently active nested names we're inside:
+Stack of the currently active nested names that are in scope for name lookup:
```